rb-files/app.rb

```diff
@@ -23,72 +19,63 @@ def list_books
     end
   end
 
-  def show_person(person)
-    print "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-    puts ", Specialization: #{person.specialization}" if person.is_a?(Teacher)
-    puts ", Classroom: #{person.classroom}" if person.instance_of?(Student)
-  end
-
   def list_people
     return puts 'No people found' if @people.empty?
 
     @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
   end
 
   def create_person
-    print 'Do you want to Create a students (1) or a teacher (2)? [Input the number]: '
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
     select_person = gets.chomp.to_i
     print 'Name: '
     name = gets.chomp
     print 'Age: '
     age = gets.chomp.to_i
     case select_person
     when 1
-      print 'Has parent permission? [Y/N]: '
-      permission = gets[0].capitalize
-      permission = permission == 'Y'
-      #   @people << Student.new(nil, age, name, permission)
-      print 'What is the student\'s classroom? '
-      classroom = gets.chomp
-      @people.push(Student.new(age, name, permission, classroom))
+      create_student(name, age)
     when 2
-      print 'Specialization: '
-      specialization = gets.chomp
-      @people << Teacher.new(age, name, specialization)
+      create_teacher(name, age)
     end
     puts 'Person created successfully'
   end
 
+  def create_student(name, age)
+    print 'Has parent permission? [Y/N]: '
+    permission = gets[0].capitalize
+    permission = permission == 'Y'
+    print 'What is the student\'s classroom? '
+    classroom = gets.chomp
+    @people.push(Student.new(age, name, permission, classroom))
+  end
+
+  def create_teacher(name, age)
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people << Teacher.new(age, name, specialization)
+  end
+
   def create_book
     print 'Title: '
     title = gets.chomp
-
     print 'Author: '
     author = gets.chomp
-
     @books << Book.new(title, author)
     puts 'Book created successfully'
   end
 
   def create_rental
     puts 'Select a book from the following list by number'
     @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
-
     book_index = gets.chomp.to_i
-
-    puts
-
-    puts 'Select a person from the following list by number (not id)'
+    puts 'Select a person from the following list by number (not ID)'
     @people.each_with_index do |person, index|
       puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
     end
-
     person_index = gets.chomp.to_i
-    puts
-
     print 'Date: '
     date = gets.chomp
```

Comment:
- [ ] The previous collector refers to this part of the code, All requests for input and input validation in your methods should've been placed in the main.rb file and/or classes/methods that are responsible for requesting or validating input. 
which means that this part of the code should be moved to a class or method which specifically handles user input.